addon/webextension/selector/shooter.js

```diff
@@ -1,28 +1,47 @@
-/* globals callBackground, documentMetadata, uicontrol, util, ui, catcher */
-/* globals XMLHttpRequest, window, location, alert, console, domainFromUrl, randomString */
+/* globals global, documentMetadata, util, uicontrol, ui, catcher */
+/* globals XMLHttpRequest, window, location, alert, domainFromUrl, randomString */
 /* globals document, setTimeout, location */
 
-window.shooter = (function () { // eslint-disable-line no-unused-vars
+"use strict";
+
+this.shooter = (function() { // eslint-disable-line no-unused-vars
   let exports = {};
   const { AbstractShot } = window.shot;
 
   const RANDOM_STRING_LENGTH = 16;
   let backend;
   let shot;
   let supportsDrawWindow;
+  const callBackground = global.callBackground;
+  const clipboard = global.clipboard;
+
+  function regexpEscape(str) {
+    // http://stackoverflow.com/questions/3115150/how-to-escape-regular-expression-special-characters-using-javascript
+    return str.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
+  }
+
+  function sanitizeError(data) {
+    const href = new RegExp(regexpEscape(window.location.href), 'g');
+    const origin = new RegExp(`${regexpEscape(window.location.origin)}[^\s",>]*`, 'g');
+    const json = JSON.stringify(data)
+      .replace(href, 'REDACTED_HREF')
+      .replace(origin, 'REDACTED_URL');
```

Comment:
This change came out of the legal/data review, though not an engineering data review.  @fmarier was looking at our data practices, but I don't think we called this part out during that, I'll ping him about this.